### PR TITLE
GH workflows refactored

### PR DESCRIPTION
**Pull request by mkuthan**
Single CI workflow for pushes to master and pull requests: 
* lint + unit-test + e2e-test
* lighthouse-report, size-report

image

In the second step I would like to move deploy workflows as a sub-workflow of CI as well. Only ci, release and undeploy brach workflows should stay top-level.

### CODE REVIEW

**Comment by mkuthan** [comment]
If smb knows why post run on composite setup action fails - let me know :)

File: .github/actions/setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Setup
+description: Setup build
+
+inputs:
+  node-version:
+    required: true
+    default: "16.15.0"
```
Comment (adrianmroz-allegro) [inline .github/actions/setup/action.yml:7]: As next step, we should bump this version thanks to #946

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup
+      - run: npm run lint
+
+  unit-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.20.0, 16.15.0]
```
Comment (adrianmroz-allegro) [inline .github/workflows/ci.yml:28]: Same here.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm run build
+      - run: npm run e2e
+
+  lighthouse-report:
+    needs: [lint, unit-test, e2e-test]
```
Comment (adrianmroz-allegro) [inline .github/workflows/ci.yml:54]: Cool stuff!

**Review by adrianmroz-allegro — APPROVED** [review]

**Pull request merged**